site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,316 @@
+
+Clients use this command to initially *add* credentials for a device that has already been registered with Hono. The credentials to be added may be of arbitrary type. However, [Standard Credential Types]({{< relref "#standard-credential-types" >}}) contains an overview of some common types that are used by Hono's protocol adapters and which may be useful to others as well.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *add credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type            | Description                         |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------------------------------- |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string*  | MUST contain the ID of the device.  |
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `add`.       |
+
+The request message MUST include payload as defined in [Request Payload]({{< relref "#request-payload" >}}).
+
+**Response Message Format**
+
+A response to an *add credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty. 
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *201* | Created, the credentials have been added successfully. |
+| *409* | Conflict, there already exist credentials with the `type` and `auth-id` provided in the request payload for the *device_id*. |
+| *412* | Precondition Failed, there is no device registered with the given *device_id* within the *tenant_id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get Credentials
+
+Protocol adapters use this command to *look up* credentials of a particular type for a device identity.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *get credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type            | Description                   |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------------------------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `get`. |
+
+The body of the message MUST consist of a single *AMQP Value* section containing a UTF-8 encoded string representation of a single JSON object as follows:
+
```

Comment:
I was thinking that here we could do like Registration API. Instead of having a JSON paylod with info about credentials to retrieve, using two custom AMQP properties : type and auth-id.
For "get registration info" we already do something like that, passing the device-id as a custom AMQP property.
@sophokles73 @dejanb wdyt ?